pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -195,66 +204,203 @@ function parseJsonFiles(rootPath: string, defaultsPath: string, lockedPath: stri
+  }
 
-  for (const k of commonKeys) {
-    const schemaVal = schemaObj[k];
-    let regValue: any = null;
+  // First process the nested keys, then process any values
+  for (const [schemaKey, registryKey] of commonKeys) {
+    const schemaVal = schemaObj[schemaKey];
 
-    if (typeof schemaVal === 'object') {
-      if (!Array.isArray(schemaVal)) {
-        const innerKey = nativeReg.openKey(regKey, k, nativeReg.Access.READ);
+    if ((typeof schemaVal) !== 'object' || schemaVal === null) {
+      const valueType = schemaVal === null ? 'null' : (typeof schemaVal);
 
-        if (!innerKey) {
-          continue;
-        }
-        try {
-          regValue = readRegistryUsingSchema(schemaVal, innerKey);
-          if (regValue && (typeof regValue === 'object') && Object.keys(regValue).length === 0) {
-            // Ignore empty objects
-            regValue = null;
-          }
-        } finally {
-          nativeReg.closeKey(innerKey);
-        }
-      } else {
-        const multiSzValue = nativeReg.queryValueRaw(regKey, k);
+      console.error(`Expecting registry entry ${ pathParts.join('\\') }\\${ registryKey } to be a ${ valueType }, but it's a registry object`);
+      continue;
+    }
+    const innerKey = nativeReg.openKey(regKey, registryKey, nativeReg.Access.READ);
 
-        if (multiSzValue) {
-          // Registry value can be a single-string or even a DWORD and parseMultiString will handle it.
-          const arrayValue = nativeReg.parseMultiString(multiSzValue as nativeReg.Value);
+    if (!innerKey) {
+      continue;
+    }
+    try {
+      regValue = readRegistryUsingSchema(schemaVal, innerKey, pathParts.concat([schemaKey]));
+    } finally {
+      nativeReg.closeKey(innerKey);
+    }
+    if (regValue) {
+      newObject ??= {};
+      (newObject as Record<string, any>)[schemaKey] = regValue;
+    }
+  }
+  for (const [schemaKey, registryKey] of specialObjectKeys) {
+    const innerKey = nativeReg.openKey(regKey, registryKey, nativeReg.Access.READ);
 
-          regValue = arrayValue.length ? arrayValue : null;
-        }
-      }
+    if (innerKey === null) {
+      console.error(`No value for registry object ${ pathParts.join('\\') }\\${ registryKey }`);
+      continue;
+    }
+    try {
+      regValue = readRegistryObject(innerKey, pathParts.concat([schemaKey]), true);
+    } catch (err: any) {
+      console.error(`Error getting registry object for ${ pathParts.join('\\') }\\${ registryKey }: `, err);
+    } finally {
+      nativeReg.closeKey(innerKey);
+    }
+    if (regValue) {
+      newObject ??= {};
+      (newObject as Record<string, any>)[schemaKey] = regValue;
+    }
+  }
+  const unknownValueNames: string[] = [];
+  const registryValueNames = nativeReg.enumValueNames(regKey);
+  const registryValueNamesFixedCase = registryValueNames.map(k => fixProfileKeyCase(k, schemaKeys));
+
+  registryValueNamesFixedCase.forEach((key, index) => {
```

Comment:
Same as above — `for (const originalName of nativeReg.enumValueNames(regKey) { const key = fixProfileKeyCase(originalName, schemaKeys);`

Reply:
And same as above, if we walk the schema instead of the registry we miss undefined entries